src/manifest.rs

```diff
@@ -165,63 +160,62 @@ impl Manifest {
     }
 
     /// Get the specified table from the manifest.
-    pub fn get_table<'a>(
-        &'a mut self,
-        table_path: &[String],
-    ) -> Result<&'a mut BTreeMap<String, toml::Value>> {
+    pub fn get_table<'a>(&'a mut self, table_path: &[String]) -> Result<&'a mut toml_edit::Item> {
         /// Descend into a manifest until the required table is found.
         fn descend<'a>(
-            input: &'a mut BTreeMap<String, toml::Value>,
+            input: &'a mut toml_edit::Item,
             path: &[String],
-        ) -> Result<&'a mut BTreeMap<String, toml::Value>> {
+        ) -> Result<&'a mut toml_edit::Item> {
             if let Some(segment) = path.get(0) {
-                let value = input
-                    .entry(segment.to_owned())
-                    .or_insert_with(|| toml::Value::Table(BTreeMap::new()));
+                let value = input[&segment].or_insert(toml_edit::table());
 
-                match *value {
-                    toml::Value::Table(ref mut table) => descend(table, &path[1..]),
-                    _ => Err(ErrorKind::NonExistentTable(segment.clone()).into()),
+                if value.is_table_like() {
+                    descend(value, &path[1..])
+                } else {
+                    Err(ErrorKind::NonExistentTable(segment.clone()).into())
                 }
             } else {
                 Ok(input)
             }
         }
 
-        descend(&mut self.data, table_path)
+
+        descend(&mut self.data.root, table_path)
     }
 
     /// Get all sections in the manifest that exist and might contain dependencies.
-    pub fn get_sections(&self) -> Vec<(Vec<String>, BTreeMap<String, toml::Value>)> {
+    /// The returned items are always `Table` or `InlineTable`.
+    pub fn get_sections(&self) -> Vec<(Vec<String>, toml_edit::Item)> {
```

Comment:
I'm not sure adding `as_table_like_mut` would make life easier.
1. We can't return `Vec<(_, Box<&mut TableLike>)>`, because it would require holding > 1 mut reference to the document.
2. Even if we return an iterator over `Box<&mut TableLike>`, we're calling `                manifest.update_table_entry(&table_path, &latest_version)?;` [here](https://github.com/killercup/cargo-edit/pull/181/files#diff-185876a000bc8529006c4808a07e8ab1R99).

As for viewing the docs, here they are: https://reusable.software/rust/toml_edit/doc/toml_edit/index.html